extra_data/keydata.py

```diff
@@ -106,6 +108,29 @@ def select_trains(self, trains):
     def __getitem__(self, item):
         return self.select_trains(item)
 
+    def split_trains(self, parts=None, trains_per_part=None):
+        """Split this data into chunks with a fraction of the trains each.
+
+        Either *parts* or *trains_per_part* must be specified.
+
+        This returns an iterator yielding new :class:`KeyData` objects.
+        The parts will have similar sizes, e.g. splitting 11 trains
+        with ``trains_per_part=8`` will produce 5 & 6 trains, not 8 & 3.
+
+        Parameters
+        ----------
+
+        parts: int
+            How many parts to split the data into. If trains_per_part is also
+            specified, this is a minimum, and it may make more parts.
+            It may also make fewer if there are fewer trains in the data.
+        trains_per_part: int
+            A maximum number of trains in each part. Parts will often have
+            fewer trains than this.
+        """
+        for s in split_trains(len(self.train_ids), parts, trains_per_part):
```

Comment:
I'd probably select only train IDs containing data for `KeyData` before splitting in parts, and maybe a `require_all` argument for the same purpose on `DataCollection`?

Reply:
Yes, I think that probably makes sense for KeyData at least.

For DataCollection, what do you think about leaving this to `.select()`? I think filtering for trains with data only makes sense when you define which data you're looking at (otherwise, some source which wasn't recording could mean no trains are included), so you'd want to select first anyway. It would look like this:

```python
for chunk in run.select('*XGM/*', require_all=True).split_trains(8):
    ...
```

Reply:
yup, I agree :+1: